epistemix_platform/src/epistemix_platform/use_cases/upload_results.py

```
@@ -0,0 +1,160 @@
+    # 6. Update run with results URL and timestamp
+    run.results_url = upload_location.url.split("?")[0]  # Remove query params (credentials)
+    run.results_uploaded_at = datetime.utcnow()
+    run.status = RunStatus.DONE  # Mark run as completed
+
+    try:
+        run_repository.save(run)
+        logger.info(
+            f"Updated run {run_id}: results_url set, status={run.status.value}, "
+            f"uploaded_at={run.results_uploaded_at.isoformat()}"
+        )
+    except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)